snuba/datasets/plans/translator/visitor.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import Generic, Optional, Sequence, TypeVar
+
+from snuba.query.expressions import (
+    Argument,
+    Column,
+    CurriedFunctionCall,
+    Expression,
+    ExpressionVisitor,
+    FunctionCall,
+    Lambda,
+    Literal,
+    SubscriptableReference,
+)
+
+TExpOut = TypeVar("TExpOut")
+
+
+@dataclass(frozen=True)
+class TranslationRules(Generic[TExpOut]):
```

Comment:
I wanted to put the MappingExpressionTranslator class at the beginning of this file. It would make this whole a lot more clear. The problem is that, while python is fine with that because of __futures__.annotations, mypy is not ok with that and it is not able to resolve the input types passed to MappingExpressionTranslator. __map_expression